shared/backups.py

```diff
@@ -559,9 +551,92 @@ def generate_public_contents():
         await ux_show_story('Failed to write!\n\n\n'+str(e))
         return
 
-    msg = '''Summary file written:\n\n%s''' % nice
+    msg = '''%s file written:\n\n%s''' % (title, nice)
     await ux_show_story(msg)
 
+async def make_summary_file(fname_pattern='public.txt'):
+    from main import dis
+
+    # record **public** values and helpful data into a text file
+    dis.fullscreen('Generating...')
+
+    # generator function:
+    body = generate_public_contents()
+
+    await write_text_file(fname_pattern, body, 'Summary')
+
+async def make_bitcoin_core_wallet(fname_pattern='bitcoin-core.txt'):
+    from main import dis, settings
+    import ustruct
+    xfp = b2a_hex(ustruct.pack('<I', settings.get('xfp'))).decode().lower()
+
+    dis.fullscreen('Generating...')
+
+    # generator function:
+    payload = ujson.dumps(generate_bitcoin_core_wallet())
+
+    body = '''\
+# Bitcoin Core Wallet Import File
+
+https://github.com/Coldcard/firmware/blob/master/docs/bitcoin-core-usage.md
+
+## For wallet with master key fingerprint: {xfp}
+
+Wallet operates on blockchain: {nb}
+
+## IMPORTANT WARNING
+
+Do **not** deposit to any address in this file unless you have a working
+wallet system that is ready to handle the funds at that address!
+
+## Bitcoin Core RPC
+
+The following command can be entered after opening Window -> Console in Bitcoin Core,
+or using bitcoin-cli:
+
+importmulti '{payload}'
+
+'''.format(payload=payload, xfp=xfp, nb=chains.current_chain().name)
+
+    await write_text_file(fname_pattern, body, 'Bitcoin Core')
+
+def generate_bitcoin_core_wallet():
+    # Generate the data for an RPC command to import keys into Bitcoin Core
+    from descriptor import AddChecksum
+    from main import settings
+    import ustruct
+
+    from public_constants import AF_P2WPKH
+
+    chain = chains.current_chain()
+    assert chain.ctype in {'BTC', 'XTN'}, "Only Bitcoin supported"
+
+    derive = "m/84'/{coin_type}'/{account}'".format(account=0, coin_type=chain.b44_cointype)
+
+    with stash.SensitiveValues() as sv:
+        xpub = chain.serialize_public(sv.derive_path(derive))
+
+    xfp = settings.get('xfp')
+    txt_xfp = b2a_hex(ustruct.pack('<I', xfp)).decode().lower()
+
+    chain = chains.current_chain()
+
+    _,vers,_ = version.get_mpy_version()
+
+    return list(map(lambda internal: {
+        'desc': AddChecksum("wpkh([{fingerprint}/84h/{coin_type}h/{account}h]{xpub}/{change}/*)".format(
+            fingerprint=txt_xfp,
+            coin_type=chain.b44_cointype,
+            account=0,
+            xpub=xpub,
+            change=1 if internal else 0
+        )),
+        'range': [0, 1000],
+        'timestamp': 'now',
```

Comment:
Does Colcard have a clock and know when the seed was generated? If so, that could be added here.

Reply:
Sadly Coldcard has no idea of time nor block height.